### PR TITLE
Add http redirects (includes http -> https redirects)

### PR DESCRIPTION
**Pull request by jkroepke**
Related: #154
Fixes: #93

I add the rediection code inside the HttpRequest Class so any http request from the launcher are able to handle redirects.

### CODE REVIEW

**Comment by hedgehog1029** [comment]
Definitely interested in merging this but there are a couple of fixes needed:
1. Code style - there's some weird spacing and such
2. The user agent was changed here unneccesarily - can this be reverted
3. Should probably handle 307, 308 and maybe 303 status codes

File: launcher/src/main/java/com/skcraft/launcher/util/HttpRequest.java [hunk 1]
```diff
@@ -109,31 +110,54 @@ public HttpRequest execute() throws IOException {
...
+            String location = null;
+            URL connectUrl = url;
 
-            if (body != null) {
-                conn.setRequestProperty("Content-Type", contentType);
-                conn.setRequestProperty("Content-Length", Integer.toString(body.length));
-                conn.setDoInput(true);
-            }
+            for(int redirect = 0; redirect <= maxRedirectFollow; redirect++) {
+                conn        = (HttpURLConnection) reformat(connectUrl).openConnection();
```
Comment (hedgehog1029) [inline launcher/src/main/java/com/skcraft/launcher/util/HttpRequest.java:117]: Weird whitespace here

File: launcher/src/main/java/com/skcraft/launcher/util/HttpRequest.java [hunk 2]
```diff
@@ -109,31 +110,54 @@ public HttpRequest execute() throws IOException {
...
 
-            for (Map.Entry<String, String> entry : headers.entrySet()) {
-                conn.setRequestProperty(entry.getKey(), entry.getValue());
-            }
+                conn.setConnectTimeout(15000);
+                conn.setReadTimeout(15000);
+                conn.setInstanceFollowRedirects(false);
+                conn.setRequestProperty("User-Agent", "SKCraft Launcher");
```
Comment (hedgehog1029) [inline launcher/src/main/java/com/skcraft/launcher/util/HttpRequest.java:122]: Can this be reverted to the previous user-agent, `"Mozilla/5.0 (Java) SKMCLauncher"`

File: launcher/src/main/java/com/skcraft/launcher/util/HttpRequest.java [hunk 3]
```diff
@@ -109,31 +110,54 @@ public HttpRequest execute() throws IOException {
...
+                    DataOutputStream out = new DataOutputStream(conn.getOutputStream());
+                    out.write(body);
+                    out.flush();
+                    out.close();
+                }
+
+                switch (conn.getResponseCode())
+                {
+                    case HttpURLConnection.HTTP_MOVED_PERM:
+                    case HttpURLConnection.HTTP_MOVED_TEMP:
```
Comment (hedgehog1029) [inline launcher/src/main/java/com/skcraft/launcher/util/HttpRequest.java:153]: Handle some other 3xx codes here

**Review by hedgehog1029 — CHANGES_REQUESTED** [review]
Obviously apply all these changes to the bootstrap copy too.

**Comment by jkroepke** [comment]
this is almost 2 year old. I guess using a high level http client (or java 11?) http client should a better option.

**Pull request closed without merging**